src/FindAnApprenticeship/SFA.DAS.FindAnApprenticeship/Application/Queries/GetAddresses/GetAddressesQueryHandler.cs

```diff
@@ -23,9 +24,10 @@ public GetAddressesQueryHandler(ILocationApiClient<LocationApiConfiguration> loc
 
         public async Task<GetAddressesQueryResult> Handle(GetAddressesQuery request, CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(request.Query)) throw new ArgumentException($"Query is required", nameof(GetAddressesQuery.Query));
+            if (string.IsNullOrEmpty(request.Postcode)) throw new ArgumentException($"Query is required", nameof(GetAddressesQuery.Postcode));
 
-            var addressesResponse = await _locationApiClient.Get<GetAddressesListResponse>(new GetAddressesQueryRequest(request.Query, _config.LocationsApiMinMatch));
+            var postcode = HttpUtility.UrlDecode(request.Postcode);
```

Comment:
you shouldnt need to do this, when it hits the controller action it will already be done